src/transformers/models/bridgetower/modeling_bridgetower.py

```diff
@@ -0,0 +1,1673 @@
+            inputs_embeds=inputs_embeds,
+            past_key_values_length=past_key_values_length,
+        )
+        encoder_outputs = self.encoder(
+            embedding_output,
+            attention_mask=extended_attention_mask,
+            head_mask=head_mask,
+            encoder_hidden_states=encoder_hidden_states,
+            encoder_attention_mask=encoder_extended_attention_mask,
+            past_key_values=past_key_values,
+            use_cache=use_cache,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+        )
+        sequence_output = encoder_outputs[0]
+        pooled_output = self.pooler(sequence_output) if self.pooler is not None else None
+
+        if not return_dict:
+            return (sequence_output, pooled_output) + encoder_outputs[1:]
+
+        return BaseModelOutputWithPoolingAndCrossAttentions(
+            last_hidden_state=sequence_output,
+            pooler_output=pooled_output,
+            past_key_values=encoder_outputs.past_key_values,
+            hidden_states=encoder_outputs.hidden_states,
+            attentions=encoder_outputs.attentions,
+            cross_attentions=encoder_outputs.cross_attentions,
+        )
+
+
+@add_start_docstrings(
+    "The bare BridgeTower Model transformer outputting BridgeTowerModelOutput object without any specific head on"
+    " top.",
+    BRIDGETOWER_START_DOCSTRING,
+)
+class BridgeTowerModel(BridgeTowerPreTrainedModel):
+    def __init__(self, config):
+        super().__init__(config)
+        self.config = config
+
+        if config.share_cross_modal_transformer_layers:
+            self.cross_modal_text_transform = nn.Linear(config.input_text_embed_size, config.hidden_size)
+            self.cross_modal_image_transform = nn.Linear(config.input_image_embed_size, config.hidden_size)
+        else:
+            self.cross_modal_text_transform = nn.ModuleList(
+                [nn.Linear(config.input_text_embed_size, config.hidden_size) for _ in range(config.num_hidden_layers)]
+            )
+            self.cross_modal_image_transform = nn.ModuleList(
+                [nn.Linear(config.input_image_embed_size, config.hidden_size) for _ in range(config.num_hidden_layers)]
+            )
+
+        self.token_type_embeddings = nn.Embedding(2, config.hidden_size)
+
+        vision_config = config.vision_config
+        self.vit_model = BridgeTowerCLIP(
+            num_hidden_layers=vision_config.num_hidden_layers,
+            hidden_size=vision_config.hidden_size,
+            patch_size=vision_config.patch_size,
+            image_size=vision_config.image_size,
+            stop_gradient=vision_config.stop_gradient,
+            share_layernorm=vision_config.share_layernorm,
+            vit_remove_last=vision_config.vit_remove_last,
+        )
+
+        text_config = config.text_config
+        self.text_transformer = BridgeTowerTextModel(text_config)
```

Comment:
Would be great to align this with the CLIP implementation [here](https://github.com/huggingface/transformers/blob/3a9bd972e22a7ef165415989b22a2055b0cf51d6/src/transformers/models/clip/modeling_clip.py#L968-L969).

In other words, have 2 modules, `BridgeTowerVisionModel` and `BridgeTowerTextModel` that just take a config as argument

Reply:
We have addressed this in our latest commits. Thanks for the suggestion!